src/services/prisma-datastore/PrismaDataProvider.ts

```diff
@@ -41,15 +93,78 @@ export class PrismaDataProvider implements DataProvider {
     private projectUID: ProjectID // todo - remove
   ) {}
 
-  private async currentProjectID(): Promise<ProjectID> {
+  async doBulkImport(
+    source?: NotehubAccessor,
+    target?: SparrowEventHandler
+  ): Promise<BulkImport> {
+    serverLogInfo("Bulk import starting");
+    const b: BulkImport = { itemCount: 0, errorCount: 0 };
+
+    if (!source)
+      throw new Error("PrismaDataProvider needs a source for bulk data import");
+    if (!target)
+      throw new Error("PrismaDataProvider needs a target for bulk data import");
+
+    const project = await this.currentProject();
+
+    // Some  details have to be fetched from the notehub api (because some
+    // gateway details like name are only available in environment variables)
+    const notehubProvider = new NotehubDataProvider(source, {
+      type: "ProjectID",
+      projectUID: project.projectUID,
+    });
+    const gateways = await notehubProvider.getGateways();
+    for (const gateway of gateways) {
+      await manageGatewayImport(b, this.prisma, project, gateway);
+    }
+
+    const nodes = await notehubProvider.getNodes(gateways.map((g) => g.uid));
+    for (const node of nodes) {
+      await manageNodeImport(b, this.prisma, project, node);
+    }
+
+    // Todo: Once this is working, remove time limit
```

Comment:
resolve this todo?

Reply:
Notehub 'Pilot' account only retains 7 days of events, so 10 days feels like plenty for bulk data import. If anyone wants to import more we could just tell them to change this. I did change it to this though so it's more descriptive.
```
    const pilotBulkImportDays = 10;
    const hoursBack = 24 * pilotBulkImportDays;
```